apps/builder/app/builder/builder.tsx

```diff
@@ -208,6 +227,72 @@ const NavigatorPanel = ({
   );
 };
 
+const revealAnimation = ({
+  show,
+  backgroundColor,
+}: {
+  show: boolean;
+  backgroundColor: string;
+}): CSS => ({
+  position: "relative",
+  "> ::after": {
+    content: "",
+    position: "absolute",
+    inset: 0,
+    zIndex: 1,
+    transitionDuration: "300ms",
+    pointerEvents: "none",
+    transitionProperty: "opacity",
+    backgroundColor,
+    opacity: show ? 0 : 1,
+  },
+});
+
+const Loading = ({ value }: { value: number }) => {
+  const [progress, setProgress] = useState(value);
+
+  // This is an approximation of a real progress that should come from "value".
+  useEffect(() => {
+    const timerId = setInterval(() => {
+      setProgress((progress) => {
+        if (progress >= 100) {
+          clearInterval(timerId);
+          return progress;
+        }
+        return progress + 1;
+      });
+    }, 50);
+    return () => {
+      clearInterval(timerId);
+    };
+  }, []);
+
+  // When real value from progress comes in, we reset our estimate.
+  useEffect(() => {
+    setProgress(value);
+  }, [value]);
+
+  if (progress === 100) {
```

Comment:
See my screenshot. Better to do progress non linear
something like https://easings.net/#easeOutExpo 
so it reaches to 80% like linear and then has long period to reach 100%